stdlib/io/src/main/java/org/ballerinalang/stdlib/io/nativeimpl/ReadJson.java

```diff
@@ -72,4 +76,22 @@ public void execute(Context context, CallableUnitCallback callback) {
     public boolean isBlocking() {
         return false;
     }
+
+    public static void readJson(Strand strand, ObjectValue channel) {
+        //TODO : TempCallableUnitCallback is temporary fix to handle non blocking call
+        TempCallableUnitCallback callback = new TempCallableUnitCallback(strand);
+
+        CharacterChannel charChannel = (CharacterChannel) channel.getNativeData(IOConstants.CHARACTER_CHANNEL_NAME);
+        CharacterChannelReader reader = new CharacterChannelReader(charChannel, new EventContext(callback));
+        final Object json;
+        try {
+            json = JSONParser.parse(reader);
+        } catch (BallerinaException e) {
```

Comment:
It seems we need to catch BallerinaException from jvm package (org.ballerinalang.jvm.util.exceptions.BallerinaException)

Reply:
Fixed via f407a897e67f368a55d46ff67fdd2899dfb11042